src/common/type-check.ts

```diff
@@ -29,5 +29,9 @@ export function isString (value: Result): boolean {
  * @return {boolean} result of the test
  */
 export function isObject (value: unknown): boolean {
+  if (value === null || value === undefined) {
```

Comment:
why not?
`return value !== null && value !== undefined && typeof value === 'object' && value.constructor === Object`

Reply:
typescript doesn't like `value.constructor` without checking for `value` as "value is possibly null",
and if we add "value" at the beginning of that chain, null && _ would always yield null.
also better readability, rule out the nil types and short-circuit

Reply:
> rule out the nil types and short-circuit
In terms of short circuiting it is exactly the same.

You could do this:
```Typescript
  return value !== null && value !== undefined && value &&
      typeof value === 'object' && value.constructor === Object
```

But that's a bit weird for me because this could be simplified to what we had before:
```Typescript
return value && typeof value === 'object' && value.constructor === Object
```

Reply:
yup that could work, and like you mentioned, it's pretty much going back to what we had before with the additional checks for the nils, i think the way it is currently written is slightly more readable because we explicitly call out the nil types and short circuit, rather than chaining 5 checks with &&.

If this really bothers you, i can change it.

Reply:
I just don't think we need the explicit check for null and undefined when are checking for the boolean value of the object later on. That adds more complexity to the code.

If I eventually need to touch this file, I would remove that if, test would pass and I would commit that change. Just trying to understand the need for it.

Reply:
https://github.com/briza-insurance/illogical/pull/44/commits/d2c47ec2a5e05f90281390250633c0c638347fc2